hydrolib/core/dflowfm/t3d/models.py

```diff
@@ -0,0 +1,319 @@
+            >>> layers = [0, 0.2, 0.6, 0.8, 1.0]
+            >>> record = [
+            ...     T3DTimeRecord(time="0 seconds since 2006-01-01 00:00:00 +00:00", data=[5.0, 5.0, 10.0, 10.0]),
+            ...     T3DTimeRecord(time="1e9 seconds since 2001-01-01 00:00:00 +00:00", data=[5.0, 5.0, 10.0, 10.0])
+            ... ]
+            >>> model = T3DModel(comments=comments, layer_type=layer_name, layers=layers, records=record)
+            >>> model.as_dict()
+            {0.0: [5.0, 5.0, 10.0, 10.0], 1000000000.0: [5.0, 5.0, 10.0, 10.0]}
+
+            ```
+        """
+        data = {}
+        for record in self.records:
+            match = TIME_PATTERN.match(record.time)
+            offset_str, _, _ = match.groups()
+            offset = float(offset_str)
+            data[offset] = record.data
+
+        return data
+
+    def get_units(self):
+        """Return the units for each quantity in the timeseries.
+
+        Returns:
+            List[str]: A list of units for each quantity in the timeseries.
+
+        Examples:
+            Create a `TimModel` object from a .tim file:
+                ```python
+                >>> from hydrolib.core.dflowfm.t3d.models import T3DTimeRecord, T3DModel
+                >>> layer_name = "SIGMA"
+                >>> comments = ["comment1", "comment2"]
+                >>> layers = [0, 0.2, 0.6, 0.8, 1.0]
+                >>> record = [
+                ...     T3DTimeRecord(time="0 seconds since 2006-01-01 00:00:00 +00:00", data=[5.0, 5.0, 10, 10, 10]),
+                ...     T3DTimeRecord(time="1e9 seconds since 2001-01-01 00:00:00 +00:00", data=[5.0, 5.0, 10, 10, 10])
+                ... ]
+                >>> model = T3DModel(comments=comments, layer_type=layer_name, layers=layers, records=record)
+                >>> model.quantities_names = ["discharge", "waterlevel", "temperature", "salinity", "initialtracer"]
+                >>> print(model.get_units())
+                ['m3/s', 'm', 'degC', '1e-3', '-']
+
+                ```
+        """
+        if self.quantities_names is None:
+            return None
+        return get_quantity_unit(self.quantities_names)
+
+
+def get_quantity_unit(quantities_names: List[str]) -> List[str]:
+    """
+    Maps each quantity in the input list to a specific unit based on its content.
+
+    Args:
+        quantities_names (list of str): A list of strings to be checked for specific keywords.
+
+    Returns:
+        list of str: A list of corresponding units for each input string.
+
+    Examples:
+        ```python
+        >>> quantities_names = ["discharge", "waterlevel", "salinity", "temperature"]
+        >>> get_quantity_unit(quantities_names)
+        ['m3/s', 'm', '1e-3', 'degC']
+
+        ```
+    """
+    # Define the mapping of keywords to units
+    unit_mapping = {
+        "discharge": "m3/s",
+        "waterlevel": "m",
+        "salinity": "1e-3",
+        "temperature": "degC",
```

Comment:
This may be correct, but I wonder: where is the change made from "salinitybnd" to "salinity"?
And: does the resulting .bc file still contain `Quantity = salinitybnd`, etc?

Reply:
well the function checks id the word `salinity` is part of the quantity name, so `anytextsalinity`, `salinityanytext`, `anytextsalinityanytext` will be assigned `1e-3`